Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -446,11 +456,30 @@ - (void)testOpenRealmWithNormalCredentials {
 
 /// If client B adds objects to a synced Realm, client A should see those objects.
 - (void)testAddObjects {
-    NSURL *url = REALM_URL();
-    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:NSStringFromSelector(_cmd)
+    [self verifyAddObjectsWithUsername:NSStringFromSelector(_cmd) relamURL:REALM_URL() simulateReconnection:NO];
+}
+
+- (void)testAddObjectsWithReconnection {
+    [self verifyAddObjectsWithUsername:NSStringFromSelector(_cmd) relamURL:REALM_URL() simulateReconnection:YES];
+}
+
+- (void)verifyAddObjectsWithUsername:(NSString *)username
+                            relamURL:(NSURL *)url
+                simulateReconnection:(BOOL)simulateReconnection {
+    NSLog(@"isParent: %d", self.isParent);
+    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:username
                                                                                             register:self.isParent]
-                                              server:[RLMObjectServerTests authServerURL]];
-    RLMRealm *realm = [self openRealmForURL:url user:user];
+                                               server:[RLMObjectServerTests authServerURL]];
+    NSLog(@"Login succeeded: %@", user.identity);
+    if (!self.isParent && simulateReconnection) {
+        [self disableNetworking];
```

Comment:
Given that in this test the child process runs before the parent process, I think instead of waiting 15 seconds to re-enable the network connectivity, maybe you should just re-enable the network connectivity right before the child version of the test returns.